report/3.Theory/ParserImplementation.tex

```diff
@@ -0,0 +1,96 @@
+\section{Parser implementation}
+
+Since the compiler language kotlin is completely interoperable with java, both SableCC and ANLTR could be obvious choices for parser if the group decided to go with a generated parser. 
+However, for the parser of HCL it was decided that since HCL's CFG is LL(k), the parser should be handwritten. 
+This is partly due to the control that a handwritten parser offers, but primarily out of curiosity and for educational purposes.
+
+Due to the nature of HCL, the parser can be implemented as a single pass parser. 
+This is because the user may not access any variables that has not yet been declared. 
+Furthermore as mentioned HCL is a LL(K) language, and it was decided that the parser should be a LL(K) recursive decent parser.
+
+Both syntactical and contextual analysis is done within the parser. 
+This means that there is no additional post parser type checking or contextual analysis step. 
+One major reason for doing in line contextual analysis is the way functions are invoked in HCL. 
+The only way to know whether anything is a function call, is by analyzing the types. 
```

Comment:
At the very least make "whether" to "if". Tho please dont refer to "anything" as "anything". What you're essentially saying (paraphrasing, but still) "The only way to know whether carrots is a function call...."